torch/_meta_registrations.py

```diff
@@ -992,111 +1098,6 @@ def meta_bmm(self, mat2):
-        q -= 1
-    return q
-
-
-def pooling_output_shape_pad_lr(
-    inputSize, kernelSize, pad_l, pad_r, stride, dilation, ceil_mode
-):
-    outputSize = (
-        div_rtn(
-            inputSize
-            + pad_l
-            + pad_r
-            - dilation * (kernelSize - 1)
-            - 1
-            + (stride - 1 if ceil_mode else 0),
-            stride,
-        )
-        + 1
-    )
-    if ceil_mode:
-        if (outputSize - 1) * stride >= inputSize + pad_l:
-            outputSize -= 1
-    return outputSize
-
-
-def pooling_output_shape(inputSize, kernelSize, pad, stride, dilation, ceil_mode):
-    check(stride != 0, lambda: "stride should not be zero")
-    check(pad >= 0, lambda: f"pad must be non-negative, but got pad: {pad}")
-    check(
-        pad <= kernelSize // 2,
-        lambda: f"pad should be at most half of kernel size, but got pad={pad} and kernel_size={kernelSize}",
-    )
-    return pooling_output_shape_pad_lr(
-        inputSize, kernelSize, pad, pad, stride, dilation, ceil_mode
-    )
-
-
-def pool2d_shape_check(
-    input,
-    kH,
-    kW,
-    dH,
-    dW,
-    padH,
-    padW,
-    dilationH,
-    dilationW,
-    nInputPlane,
-    inputHeight,
-    inputWidth,
-    outputHeight,
-    outputWidth,
-    memory_format,
-):
-    ndim = input.dim()
-    nOutputPlane = nInputPlane
-
-    check(
-        kW > 0 and kH > 0,
-        lambda: "kernel size should be greater than zero, but got kH: {kH}, kW: {kW}",
-    )
-    check(
-        dW > 0 and dH > 0,
-        lambda: "stride should be greater than zero, but got dH: {dH}, dW: {dW}",
-    )
-    check(
-        dilationH > 0 and dilationW > 0,
-        lambda: "dilation should be greater than zero, but got dilationH: {dilationH}, dilationW: {dilationW}",
-    )
-
-    valid_dims = input.size(1) != 0 and input.size(2) != 0
-
-    if memory_format == torch.channels_last:
-        check(
-            ndim == 4 and valid_dims and input.size(3) != 0,
-            lambda: "Expected 4D (batch mode) tensor expected for input with channels_last layout"
-            " with optional 0 dim batch size for input, but got: {input.size()}",
-        )
-    else:
-        check(
-            (ndim == 3 and input.size(0) != 0 and valid_dims)
-            or (ndim == 4 and valid_dims and input.size(3) != 0),
-            lambda: f"Expected 3D or 4D (batch mode) tensor with optional 0 dim batch size for input, but got: {input.size()}",
-        )
-
-    check(
-        kW // 2 >= padW and kH // 2 >= padH,
-        lambda: "pad should be smaller than or equal to half of kernel size, but got "
-        f"padW = {padW}, padH = {padH}, kW = {kW}, kH = {kH}",
-    )
-
-    check(
-        outputWidth >= 1 and outputHeight >= 1,
-        lambda: f"Given input size: ({nInputPlane}x{inputHeight}x{inputWidth}). "
-        f"Calculated output size: ({nOutputPlane}x{outputHeight}x{outputWidth}). "
-        "Output size is too small",
-    )
-
-
```

Comment:
The code motion makes this diff annoying to review. Can we move them back

Reply:
Yes. Moved back. Sorry about that.